### PR TITLE
itest: resume pending package send

### PR DESCRIPTION
**Pull request by ffranr**
This PR adds an itest which ensure that we resume the send procedure which was interrupted by a tapd restart.

### CODE REVIEW

File: itest/send_test.go [hunk 1]
```diff
@@ -127,6 +127,98 @@ func testBasicSend(t *harnessTest) {
...
+		},
+	)
+	require.NoError(t.t, err)
+	assertAddrCreated(t.t, recvTapd, rpcAssets[0], recvAddr)
+
+	// We will now start two asset send events in sequence. We will stop and
+	// restart the sending node during each send. During one sending event
+	// we will mine whilst the sending node is stopped. During the other
+	// sending event we will only mine once the sending node is restarted.
+	for i := range []int{0, 1} {
```
Comment (guggero) [inline itest/send_test.go:176]: This works as intended, even though `i` will contain the _index_, not the slice values. Could also just use the c-like syntax: `for i := 0; i < 2; i++ {`.

**Review by guggero — APPROVED** [review]
Nice test, LGTM :tada:

**Review by jharveyb — APPROVED** [review]
Nice to have this in (and that restart works!) 🚀

File: itest/send_test.go (hunk 1 above)
Comment (ffranr) [reply itest/send_test.go:176]: Will fix in a future PR, thanks.

Comment (ffranr) [reply itest/send_test.go:176]: Fixed in this PR: https://github.com/lightninglabs/taproot-assets/pull/371

**Pull request merged**